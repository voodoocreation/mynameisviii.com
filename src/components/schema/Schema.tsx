import * as React from "react";

interface IProps {
  [index: string]: any;
  "@context"?: string;
  "@type"?: string;
  isPretty?: boolean;
}

const Schema: React.FC<IProps> = ({ isPretty, ...props }) => (
  <script
    dangerouslySetInnerHTML={{
      __html: JSON.stringify(props, null, isPretty ? "  " : undefined),
    }}
    type="application/ld+json"
  />
);

Schema.defaultProps = {
  "@context": "http://schema.org/",
  "@type": "Thing",
  isPretty: false,
};

export default Schema;
