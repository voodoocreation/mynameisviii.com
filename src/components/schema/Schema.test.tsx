import { Wrapper } from "react-test-wrapper";

import Schema from "./Schema";

const component = new Wrapper(Schema);

describe("[schema] <Schema />", () => {
  it("renders correctly", () => {
    const wrapper = component.render();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly when isPretty=true", () => {
    const wrapper = component
      .withProps({
        isPretty: true,
      })
      .render();

    expect(wrapper).toMatchSnapshot();
  });
});
