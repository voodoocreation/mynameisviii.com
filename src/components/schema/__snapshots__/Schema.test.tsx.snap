// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[schema] <Schema /> renders correctly 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"http://schema.org/","@type":"Thing"}
</script>
`;

exports[`[schema] <Schema /> renders correctly when isPretty=true 1`] = `
<script
  type="application/ld+json"
>
  {
  "@context": "http://schema.org/",
  "@type": "Thing"
}
</script>
`;
